### PR TITLE
s3ext: refactor Makefiles

### PR DESCRIPTION
**Pull request by adam8157**
Put Makefile in each subdirectory, instead of using one Makefile at top
directory to do all the things, and extract the common parts.

Signed-off-by: Adam Lee <ali@pivotal.io>
Signed-off-by: Haozhou Wang <hawang@pivotal.io>

### CODE REVIEW

File: gpAux/extensions/gps3ext/include/makefile.in [hunk 1]
```diff
@@ -0,0 +1,7 @@
+COMMON_OBJS = s3conf.o s3common.o s3utils.o s3log.o s3url_parser.o s3http_headers.o s3interface.o s3restful_service.o uncompress_reader.o s3key_reader.o s3bucket_reader.o #gpreader.o s3extbase.o
```
Comment (yydzero) [inline gpAux/extensions/gps3ext/include/makefile.in]: What is #gpreader.o? why there is '#'   comment out all remaining files?

File: gpAux/extensions/gps3ext/include/makefile.in [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+COMMON_LINK_OPTIONS = -lstdc++ -lxml2 -lpthread -lcrypto -lcurl -lz
+
+COMMON_CPP_FLAGS = -O2 -std=c++98 -Wall -fPIC -I/usr/include/libxml2
+
+TEST_OBJS = $(patsubst %.o,%_test.o,$(COMMON_OBJS))
```
Comment (yydzero) [inline gpAux/extensions/gps3ext/include/makefile.in]: cool trick to use more Makefile features!

File: gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Include
```
Comment (yydzero) [inline gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile]: shall we make it into bin/ directory?  otherwise gpcheckcloud might have only one file.

**Comment by yydzero** [comment]
Do we need Makefile under src directory?  Could not see it in this patch.

Others LGTM

File: gpAux/extensions/gps3ext/include/makefile.in (hunk 1 above)
Comment (adam8157) [inline gpAux/extensions/gps3ext/include/makefile.in]: It's broken in this branch, we are working on the gpreader's refactor.

File: gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile (hunk 3 above)
Comment (adam8157) [inline gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile]: bin/ should only contain binary files I suppose, or rename it to bin/ once we get another binary object.

**Comment by adam8157** [comment]
@yydzero .so module's Makefile is still at the root directory because I think it is the main compiling object, better to move it to src/ ?

File: gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile (hunk 3 above)
Comment (kuien) [inline gpAux/extensions/gps3ext/src/gpcheckcloud/Makefile]: I think gps3ext/src/gpcheckcloud/gpcheckcloud.cpp is too deep to find,  may we keep it in gps3ext/utils/gpcheckcloud.cpp? we will implement more and more tools in future.

**Comment by kuien** [comment]
LGTM. Look pretty then before, pls merge it.

**Comment by yydzero** [comment]
utils is better, or could just put it under src directly.

On Fri, Jun 17, 2016 at 7:06 AM, Kuien Liu <notifications@github.com> wrote:

> LGTM. Look pretty then before, pls merge it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/adam8157/gpdb/pull/10#issuecomment-226639151>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAfRk6dnNYSkoX7t5a4KQ0jAG0-t955Rks5qMdcJgaJpZM4I3G_f>
> .
>



-- 
Best Regards,
Yandong

**Pull request merged**